src/components/profileDatasetList.tsx

```diff
@@ -5,18 +5,31 @@ import * as api from '@/adapters/api'
 import { useContext, useEffect, useState } from 'react'
 import { authContext } from '@/lib/context/authContext'
 import { datasetWithRelationsSchema } from '@/types/zod'
+import { getUserFromURL } from '@/lib/utils'
+import { usePathname } from 'next/navigation'
 
 export default function ProfileDatasetList() {
+    const path = usePathname()
+    const userNameURL = getUserFromURL(path)
```

Comment:
I'm a bit unsure about the need of this function. I would rather pass in props to the components on how it should fetch data.

the username slug should come from the param, either pass it down from the page or use something like
https://nextjs.org/docs/app/api-reference/functions/use-params